pkg/devfile/parse.go

```diff
@@ -24,3 +24,23 @@ func Parse(path string) (d devfileParser.DevfileObj, err error) {
 	// Successful
 	return d, nil
 }
+
+// ParseInMemory func parses and validates the devfile integrity.
+// Creates devfile context and runtime objects
+func ParseInMemory(data []byte) (d devfileParser.DevfileObj, err error) {
```

Comment:
Seems like it would make sense to make all file/path based functions extract their bytes and then call the "in-memory" functions instead of more or less duplicating code at several levels to do the same kind of thing.

Reply:
I tried doing this but found that some commands ran a lot slower as we would have to download the file to memory every time :)

Reply:
OK, let's not do that right now then, I'll take a look once this is merged.